be/src/service/staros_worker.cpp

```diff
@@ -43,6 +43,16 @@ StatusOr<staros::starlet::ShardInfo> StarOSWorker::get_shard_info(ShardId id) {
     return it->second;
 }
 
+Status StarOSWorker::iterate_shard(const std::function<bool(const ShardInfo&)>& cb) {
+    std::lock_guard l(_mtx);
+    for (const auto& shardinfo : _shards) {
+        if (!cb(shardinfo.second)) {
+            break;
+        }
+    }
+    return Status::OK();
+}
+
```

Comment:
How about returning a copy of `_shards`:
```
std::vector<ShardInfo> shards() {
    std::lock_guard l(_mtx);
    return _shards;
}
```

Reply:
This requires evaluating the memory consumption and the calling frequency of the interface. According to the current implementation (perhaps more fields will be added to shardinfo later), for memory, a single shardinfo currently occupies about 80 bytes, and 100,000 shardinfo consumes 8M of memory. If this interface is not called very frequently, it looks fine

Reply:
This interface will only be called by GC, which should be infrequent.